test/e2e/federated-service.go

```diff
@@ -340,11 +317,45 @@ func discoverService(f *framework.Framework, name string) {
 		if status.State.Terminated == nil {
 			return logerr(fmt.Errorf("container is not in terminated state"))
 		}
-		if status.State.Terminated.ExitCode == 0 {
+		if status.State.Terminated.ExitCode == code {
 			return nil
 		}
 
 		return logerr(fmt.Errorf("exited %d", status.State.Terminated.ExitCode))
-	}, time.Minute*2, time.Second*2).
-		Should(BeNil(), "%q should exit 0, but it never did", command)
+	}
+}
+
+func discoverService(f *framework.Framework, name string, exists bool) {
+	command := []string{"sh", "-c", fmt.Sprintf("until nslookup '%s'; do sleep 1; done", name)}
+	By(fmt.Sprintf("Looking up %q", name))
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:   FederatedServicePod,
+			Labels: map[string]string{"name": FederatedServicePod},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:    "federated-service-discovery-container",
+					Image:   "gcr.io/google_containers/busybox:1.24",
+					Command: command,
+				},
+			},
+			RestartPolicy: api.RestartPolicyOnFailure,
+		},
+	}
+
+	_, err := f.Client.Pods(f.Namespace.Name).Create(pod)
```

Comment:
Use the versioned client now that we have that?
(We aim to eliminate all uses of unversioned client and remove them)